types/chrome-remote-interface/index.d.ts

```diff
@@ -0,0 +1,41 @@
+// Type definitions for chrome-remote-interface 0.30
+// Project: https://github.com/cyrus-and/chrome-remote-interface
+// Definitions by: Khairul Azhar Kasmiran <https://github.com/kazarmy>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import ProtocolProxyApi from './devtools-protocol/protocol-proxy-api';
+
+declare namespace CDP {
+    interface BaseOptions {
+        host?: string;
+        port?: number;
+    }
+
+    interface ListJson {
+        description: string;
+        devtoolsFrontendUrl: string;
+        id: string;
+        title: string;
+        type: string;
+        url: string;
+        webSocketDebuggerUrl: string;
+    }
+
+    interface Client {
+        close: () => Promise<void>;
+        Page: ProtocolProxyApi.PageApi;
+        Runtime: ProtocolProxyApi.RuntimeApi;
+    }
+}
+
+declare const CDP: {
+    (options?: CDP.BaseOptions): Promise<CDP.Client>;
```

Comment:
Should there be a `CDP` version with callback function?:
https://unpkg.com/browse/chrome-remote-interface@0.30.1/index.js

Reply:
Yes 53e2027 sorry oversight ... was exclusively looking at a different file ([lib/devtools.js](https://github.com/cyrus-and/chrome-remote-interface/blob/v0.30.1/lib/devtools.js)) than the one above.

The callback type definition is based on https://github.com/cyrus-and/chrome-remote-interface/blob/v0.30.1/index.js#L19. I don't see an attachment to the `error` event, just [a comment](https://github.com/cyrus-and/chrome-remote-interface/blob/v0.30.1/index.js#L15) stating that the error callback can be registered later. Not sure how connection errors (e.g. incorrect port) can be handled though if so.